### PR TITLE
Certain key/values are interpreted in YAML and have special meaning

### PR DESCRIPTION
**Pull request by ereslibre**
For example, Norway country that we represent with its alpha-2 ISO
code will be `NO`, which will get translated when loading this YAML
into memory as False.

Force literal values and we avoid this problem.

Fixes: bsc#1039730

### CODE REVIEW

**Comment by flavio** [comment]
LGTM, is there any way to cover that with some tests?

File: salt/cert/init.sls [hunk 1]
```diff
@@ -38,14 +38,14 @@ include:
...
-    - SN: {{ pillar['certificate_information']['subject_properties']['SN'] }}
-    - ST: {{ pillar['certificate_information']['subject_properties']['ST'] }}
+    - C: "{{ pillar['certificate_information']['subject_properties']['C'] }}"
+    - Email: "{{ pillar['certificate_information']['subject_properties']['Email'] }}"
+    - GN: "{{ pillar['certificate_information']['subject_properties']['GN'] }}"
+    - L: "{{ pillar['certificate_information']['subject_properties']['L'] }}"
+    - O: "{{ pillar['certificate_information']['subject_properties']['O'] }}"
+    - OU: "{{ pillar['certificate_information']['subject_properties']['OU'] }}"
+    - SN: "{{ pillar['certificate_information']['subject_properties']['SN'] }}"
+    - ST: "{{ pillar['certificate_information']['subject_properties']['ST'] }}"
```
Comment (inercia) [inline salt/cert/init.sls]: I think all these pillar values (and, in general, all the things that come from a user input) would need to be "yaml escaped" with something like `yaml_encode` or `yaml_dquote` (see [this](https://docs.saltstack.com/en/latest/topics/jinja/index.html#yaml_dquote)).

**Comment by ereslibre** [comment]
> LGTM, is there any way to cover that with some tests?

We would need to add this to the e2e-tests, since this is how salt interprets the `NO` without quotes.

File: salt/cert/init.sls (hunk 1 above)
Comment (ereslibre) [inline salt/cert/init.sls]: We aren't embedding jinja into jinja... it's the fact that `NO` gets interpreted by the yaml parser as `False`, so quoting is the optimal solution IMHO.

Comment (inercia) [inline salt/cert/init.sls]: `yaml_dquote` is not for embeding YAML code but for escaping some text and produce some valid YAML code. From the documentation: `yaml_dquote`: _"Serializes a string into a properly-escaped YAML double-quoted string. This is useful when the contents of a string are unknown and may contain quotes or unicode that needs to be preserved. The resulting string will be emitted with opening and closing double quotes."_

Not strictly necessary in this case (as you said, `"` should be enough), but it also escapes unicode and so...

Comment (ereslibre) [inline salt/cert/init.sls]: Indeed, `yaml_dquote` can be used. I wonder if it can fix bsc#1031475

Comment (ereslibre) [inline salt/cert/init.sls]: I am not sure if in this case it would detect `NO` and quote it... might be the case. It's worth checking it.

**Pull request merged**